op-node/eth/receipts.go

```diff
@@ -0,0 +1,54 @@
+package eth
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// EncodeReceipts encodes a list of receipts into raw receipts. Some non-consensus meta-data may be lost.
+func EncodeReceipts(elems []*types.Receipt) ([]hexutil.Bytes, error) {
+	out := make([]hexutil.Bytes, len(elems))
+	for i, el := range elems {
+		dat, err := el.MarshalBinary()
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal receipt %d: %w", i, err)
+		}
+		out[i] = dat
+	}
+	return out, nil
+}
+
+// DecodeRawReceipts decodes receipts and adds additional blocks metadata.
+// The contract-deployment addresses are not set however (high cost, depends on nonce values, unused by op-node).
```

Comment:
I know we don't use everything in the receipt, but I wonder if we should migrate to a different type (like `ConsensusReceipt`) and use that inside the op-node b/c it worries be that there is a difference here.

Reply:
That's a good idea for a larger refactor. Particularly if we can just use the consensus data and not have to fill in extra data from the transactions etc.

Reply:
`ConsensusReceipt` sounds like a good idea, but for now I just generalized the function that the client in the op-node uses when fetching raw receipts (from the geth debug API)